python/lsst/cell_coadds/_fits.py

```diff
@@ -0,0 +1,407 @@
+
+        identifiers = CellIdentifiers(
+            cell=Index2D(data["cell_id"][0], data["cell_id"][1]),
+            skymap=common.identifiers.skymap,
+            tract=common.identifiers.tract,
+            patch=common.identifiers.patch,
+            band=common.identifiers.band,
+        )
+
+        return SingleCellCoadd(
+            outer=image_planes,
+            psf=psf,
+            inner_bbox=Box2I(
+                minimum=Point2I(
+                    header["GRCELL1"] * data["cell_id"][0],
+                    header["GRCELL2"] * data["cell_id"][1],
+                ),
+                maximum=Point2I(
+                    header["GRCELL1"] * (data["cell_id"][0] + 1) - 1,
+                    header["GRCELL2"] * (data["cell_id"][1] + 1) - 1,
+                ),
+            ),
+            common=common,
+            identifiers=identifiers,
+            inputs=None,  # type: ignore[arg-type]  # TODO: Pass a sensible value here.
+        )
+
+    # TODO: Make this reader a context manager that handles file closures.
+
+    def readBBox(self) -> Box2I:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readPsf(self) -> ImageF:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readFilter(self) -> str:
+        """Read the filter information from the FITS file.
+
+        Returns
+        -------
+        band : `str`
+            The filter information read from the FITS file.
+        """
+        with fits.open(self.filename) as hdu_list:
+            band = hdu_list[1].header["BAND"]
+        return band
+
+    def readXY0(self) -> Point2I:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readWcs(self) -> afwGeom.SkyWcs:
+        """Read the WCS information from the FITS file.
+
+        Returns
+        -------
+        wcs : `~lsst.afw.geom.SkyWcs`
+            The WCS information read from the FITS file.
+        """
+        # Read in WCS
+        ps = PropertySet()
+        with fits.open(self.filename) as hdu_list:
+            ps.update(hdu_list[0].header)
+        wcs = afwGeom.makeSkyWcs(ps)
+        return wcs
+
+
+def writeMultipleCellCoaddAsFits(
+    multiple_cell_coadd: MultipleCellCoadd,
+    filename: str,
+    overwrite: bool = False,
+    metadata: PropertySet | None = None,
+) -> None:
+    """Write a MultipleCellCoadd object to a FITS file.
+
+    Parameters
+    ----------
+    multiple_cell_coadd : `MultipleCellCoadd`
+        The multiple cell coadd to write to a FITS file.
+    filename : `str`
+        The name of the file to write to.
+    overwrite : `bool`, optional
+        Whether to overwrite the file if it already exists?
+    """
+    array = [cell.identifiers.cell for cell in multiple_cell_coadd.cells.values()]
+    cell_id = fits.Column(name="cell_id", format="2I", array=array)
+
+    array = [cell.outer.image.array for cell in multiple_cell_coadd.cells.values()]
```

Comment:
I think it'd be a bit better to call this `image_array` (and the previous one `cell_id_array`) and avoid variable-name reuse here.  Or just put the array creation directly in the `Column` constructor call.

Reply:
I intentionally went for a variable name reuse in case gc decided to kick in. I'll move this to `Column` constructor call for now, and this would perhaps go away anyway if we switch to `fitstio`.

Reply:
I went with calling `image_array` and similar names for others, because I need to specify the size and dimensions explicitly, and couldn't create the array in the constructor call directly.